js/sign/src/cli.ts

```diff
@@ -24,10 +26,22 @@ function readOptions() {
     .parse(process.argv);
 }
 
-export function main() {
+async function parseMaybeEncryptedKey(
+  privateKeyFile: string
+): Promise<KeyObject> {
+  try {
+    return parsePemKey(privateKeyFile);
+  } catch (e) {
+    console.warn("This might be an encrypted private key, let's try again.");
+  }
+
+  return parsePemKey(privateKeyFile, await readPassphrase());
```

Comment:
Could you use `e.code` in the catch block to check if it is indeed just missing the passphrase, and otherwise re-throw the error?

Pseudocode:
```
try {
  read();
} catch (e) {
  if (e.code !== 'ERR_MISSING_PASSPHRASE') {
    throw e;
  }
  read(await passphrase());
}
```

https://nodejs.org/api/errors.html#err_missing_passphrase

Reply:
As discussed offline the e.code is inconsistent between node versions so keeping as is for now